ureq 2.9.3 fails to compile with errors in `hoot`

I use ureq as `ureq = { version = "2.9.3", features = ["json"] }` and compilation with Rust v1.73.0 fails with the following error.

Repro: https://github.com/msfjarvis/healthchecks-rs/pull/46

```
error[E0445]: crate-private trait `types::Private` in public interface
  --> /home/msfjarvis/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hoot-0.1.2/src/types.rs:10:1
   |
7  | pub(crate) trait Private {
   | ------------------------ `types::Private` declared as crate-private
...
10 | pub trait State: Private {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^ can't leak crate-private trait

error[E0445]: crate-private trait `types::Private` in public interface
  --> /home/msfjarvis/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hoot-0.1.2/src/types.rs:12:1
   |
7  | pub(crate) trait Private {
   | ------------------------ `types::Private` declared as crate-private
...
12 | pub trait Version: Private {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak crate-private trait

error[E0445]: crate-private trait `types::Private` in public interface
  --> /home/msfjarvis/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hoot-0.1.2/src/types.rs:19:1
   |
7  | pub(crate) trait Private {
   | ------------------------ `types::Private` declared as crate-private
...
19 | pub trait Method: Private {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak crate-private trait

error[E0445]: crate-private trait `types::Private` in public interface
  --> /home/msfjarvis/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hoot-0.1.2/src/types.rs:42:1
   |
7  | pub(crate) trait Private {
   | ------------------------ `types::Private` declared as crate-private
...
42 | pub trait BodyType: Private {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak crate-private trait

error[E0446]: crate-private type `Writer<'_, '_>` in public interface
  --> /home/msfjarvis/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hoot-0.1.2/src/header.rs:60:1
   |
60 | / pub fn check_and_output_header(
61 | |     mut w: Writer,
62 | |     version: HttpVersion,
63 | |     name: &str,
64 | |     bytes: &[u8],
65 | |     trailer: bool,
66 | | ) -> Result<()> {
   | |_______________^ can't leak crate-private type
   |
  ::: /home/msfjarvis/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hoot-0.1.2/src/out.rs:42:1
   |
42 |   pub(crate) struct Writer<'b, 'a> {
   |   -------------------------------- `Writer<'_, '_>` declared as crate-private

error[E0446]: private type `CallState` in public interface
   --> /home/msfjarvis/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hoot-0.1.2/src/body.rs:10:1
    |
10  | / pub fn do_read_body<'a, 'b>(
11  | |     state: &mut CallState,
12  | |     src: &'a [u8],
13  | |     dst: &'b mut [u8],
14  | | ) -> Result<BodyPart<'b>> {
    | |_________________________^ can't leak private type
    |
   ::: /home/msfjarvis/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hoot-0.1.2/src/lib.rs:106:1
    |
106 |   pub(crate) struct CallState {
    |   --------------------------- `CallState` declared as private

Some errors have detailed explanations: E0445, E0446.
For more information about an error, try `rustc --explain E0445`.
error: could not compile `hoot` (lib) due to 6 previous errors
warning: build failed, waiting for other jobs to finish...
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -41,6 +41,27 @@ jobs:
         with:
           command: doc
           args: --no-deps --all-features --document-private-items
+
+  build_versions:
+    strategy:
+      matrix:
+        rust: [stable, beta, 1.61.0]
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.rust }}
+          override: true
+      - uses: Swatinem/rust-cache@v1
+      - uses: actions-rs/cargo@v1
+        with:
+          command: build
+      - uses: actions-rs/cargo@v1
+        with:
+          command: build
+
   build_and_test:
     name: Test
     runs-on: ubuntu-latest

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -59,11 +59,11 @@ http-02 = { package = "http", version = "0.2", optional = true }
 http = { version = "1.0", optional = true }
 
 # This can't be in dev-dependencies due to doc tests.
-hootbin = { version = "0.1.0" }
+hootbin = { version = "0.1.1" }
 
 [dev-dependencies]
 serde = { version = "1", features = ["derive"] }
-env_logger = "0.11.1"
+env_logger = "<=0.9"
 rustls = { version = "0.22.0" }
 rustls-pemfile = { version = "2.0" }
 

--- a/src/header.rs
+++ b/src/header.rs
@@ -152,7 +152,7 @@ impl Header {
 }
 
 /// For non-utf8 headers this returns [`None`] (use [`get_header_raw()`]).
-pub fn get_header<'h>(headers: &'h [Header], name: &str) -> Option<&'h str> {
+pub(crate) fn get_header<'h>(headers: &'h [Header], name: &str) -> Option<&'h str> {
     headers
         .iter()
         .find(|h| h.is_name(name))
@@ -160,14 +160,14 @@ pub fn get_header<'h>(headers: &'h [Header], name: &str) -> Option<&'h str> {
 }
 
 #[allow(unused)]
-pub fn get_header_raw<'h>(headers: &'h [Header], name: &str) -> Option<&'h [u8]> {
+pub(crate) fn get_header_raw<'h>(headers: &'h [Header], name: &str) -> Option<&'h [u8]> {
     headers
         .iter()
         .find(|h| h.is_name(name))
         .map(|h| h.value_raw())
 }
 
-pub fn get_all_headers<'h>(headers: &'h [Header], name: &str) -> Vec<&'h str> {
+pub(crate) fn get_all_headers<'h>(headers: &'h [Header], name: &str) -> Vec<&'h str> {
     headers
         .iter()
         .filter(|h| h.is_name(name))
@@ -175,11 +175,11 @@ pub fn get_all_headers<'h>(headers: &'h [Header], name: &str) -> Vec<&'h str> {
         .collect()
 }
 
-pub fn has_header(headers: &[Header], name: &str) -> bool {
+pub(crate) fn has_header(headers: &[Header], name: &str) -> bool {
     get_header(headers, name).is_some()
 }
 
-pub fn add_header(headers: &mut Vec<Header>, header: Header) {
+pub(crate) fn add_header(headers: &mut Vec<Header>, header: Header) {
     let name = header.name();
     if !name.starts_with("x-") && !name.starts_with("X-") {
         headers.retain(|h| h.name() != name);

